CuratedSyncExamples/demo/src/main/java/io/realm/curatedsyncexamples/fieldencryption/ext/EncryptionExt.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.curatedsyncexamples.fieldencryption.ext
+
+import io.realm.curatedsyncexamples.fieldencryption.models.SerializableCipherSpec
+import io.realm.curatedsyncexamples.fieldencryption.models.SerializableSecretKey
+import io.realm.curatedsyncexamples.fieldencryption.models.UserKeyStore
+import io.realm.kotlin.mongodb.User
+import java.security.Key
+import java.security.MessageDigest
+import javax.crypto.KeyGenerator
+
+const val HASH_ALGORITHM = "SHA-256"
+
+fun Key.computeHash(): ByteArray =
+    MessageDigest.getInstance(HASH_ALGORITHM).digest(encoded)
+
+suspend fun UserKeyStore.getKeyOrGenerate(
```

Comment:
Why are these extension methods? 

Also, it would be nice with some documentation for these methods.

Reply:
No longer applicable, we have removed the UserKeyStore.